near-sdk/tests/store_performance_tests.rs

```diff
@@ -0,0 +1,268 @@
+// As wasm VM performance is tested, there is no need to test this on other types of OS.
+// This test runs only on Linux, as it's much slower on OS X due to an interpreted VM.
+#![cfg(target_os = "linux")]
+
+use near_account_id::AccountId;
+use near_gas::NearGas;
+use near_workspaces::network::Sandbox;
+use near_workspaces::types::{KeyType, SecretKey};
+use near_workspaces::{Account, Worker};
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+use strum::IntoEnumIterator;
+use strum_macros::{Display, EnumIter};
+use tokio::task::JoinSet;
+
+#[derive(Serialize, Deserialize, EnumIter, Display, Copy, Clone, PartialEq, Eq, Hash)]
+#[serde(crate = "near_sdk::serde")]
+pub enum Collection {
+    IterableSet,
+    IterableMap,
+    UnorderedSet,
+    UnorderedMap,
+    LookupMap,
+    LookupSet,
+    TreeMap,
+    Vector,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(crate = "near_sdk::serde")]
+pub enum Op {
+    Insert(u32),
+    Remove(u32),
+    Flush,
+    Contains(u32),
+    Iter(usize),
+}
+
+fn random_account_id(collection: Collection, seed: &str) -> AccountId {
+    let mut rng = rand::thread_rng();
+    let random_num = rng.gen_range(10000000000000usize..99999999999999);
+    let account_id = format!(
+        "dev-{}-{}-{}-{}",
+        random_num,
+        seed,
+        random_num,
+        collection.to_string().to_lowercase()
+    );
+    let account_id: AccountId =
+        account_id.try_into().expect("could not convert dev account into AccountId");
+
+    account_id
+}
+async fn dev_generate(
+    worker: Arc<Worker<Sandbox>>,
+    collection: Collection,
+    seed: String,
+) -> anyhow::Result<(Account, Collection)> {
+    let id = random_account_id(collection, &seed);
+    let sk = SecretKey::from_seed(KeyType::ED25519, &seed);
+    let account = worker.create_tla(id.clone(), sk).await?;
+    Ok((account.into_result()?, collection))
+}
+#[tokio::test]
+#[allow(clippy::needless_range_loop)]
+// Note that different types of tests are executed sequentially, as the previous tests are populating
+// the data for the next ones.
+// Also, the tests for each collection are executed sequentially, as otherwise near-sandbox doesn't cope.
+async fn combined_test() -> anyhow::Result<()> {
+    let worker = Arc::new(near_workspaces::sandbox().await?);
+    let contract = worker.dev_deploy(include_bytes!("test-contracts/store/res/store.wasm")).await?;
+    let res = contract.call("new").max_gas().transact().await?;
+    assert!(res.is_success());
+    let contract_id = contract.id().clone();
+    let mut account_pool = Vec::new();
+
+    // Generate different accounts to avoid Nonce collisions when executing transactions in parallel.
+    // We could generate this per collection as well, but parallel execution for those proved to be
+    // too much for the sandbox, so we're keeping it simple.
+    for val in 0..=17 {
```

Comment:
This magic constant obviously depends on something below. It is really hard to follow this code.